### ISSUE TITLE
Add recording MBID to track metadata

### ISSUE THREAD

**Issue by lyarenei** [jellyfin/jellyfin-plugin-musicbrainz#7]
Hi, if I understand correctly, this plugin is the future final destination of all musicbrainz stuff in the Jellyfin server.

So, I want to know... Is it possible to add a completely new metadata (recording MBID) to the server via this plugin?
I think it's not possible (or at least not yet), because there needs to be a new column in the database and also new field is to be displayed in the metadata editor - only possible by adding the code in the server/web directly.

Or is it actually possible to do it via plugin? Or are my assumptions correct?

If my assumptions are correct - do you have any rough guess what one needs to do to accomplish this? If this would be something simple, I could even tackle this, but if not then I'm afraid this is something out of my reach for now.

**Comment by crobibero** [comment]
What exactly needs to be added? I've never heard of "recording MBID" before. 
If new database fields are needed, you are correct that the plugin cannot add them. If we were to add a new field, a plugin would be able to provide metadata for it, but that will not happen before the library database is rewritten.

**Comment by lyarenei** [comment]
> What exactly needs to be added? I've never heard of "recording MBID" before.
I'm talking about [Recording MBID](https://musicbrainz.org/doc/Recording). It's more precisely explained there, but the generala idea is that you have a single track, which is released from a recording. A recording can be created for example by simply copying the original audio source or by mixing multiple audio sources together.

What I'd want to have in Jellyfin is exactly that - another track metadata entry for this MBID.

> If new database fields are needed, you are correct that the plugin cannot add them. If we were to add a new field, a plugin would be able to provide metadata for it, but that will not happen before the library database is rewritten.

I see. Well that's unfortunate. I could probably look into that to see how much work would that need, but I think it'd be better to wait for the database rewrite before trying anything.

**Comment by lyarenei** [comment]
> I think what @lyarenei is referring to is listed under "MusicBrainz Track Id" in the external id section of the track metadata.

@tuxfoo No, track MBID is completely different from a recording MBID. Please take a look at the documentation I linked, there is a nice chart which illustrates relations between audio, release and a track. Basically a recording is just that - a recording of an audio. You then do some post-processing on the recording, which may be then released as a track.

---
Now, my understandings of Jellyfin plugins got much better over these two, almost three years and when I (re)-implemented an Apple Music metadata provider plugin, I did not realize (until now) that this can be (relatively) simply fixed by implementing this missing metadata provider in the plugin. As the recording MBID is not read from the files by the server during scanning, I can just make the requests to MusicBrainz API, just like the ListenBrainz plugin does. Doing it this way, it wouldn't be necessary to make changes to Jellyfin DB, at least until the infrastructure is ready and reading from the file could be easily implemented.

Also, if I understand correctly, all music videos on MusicBrainz seem to be categorized as a recording? So implementing a recording MBID provider should also address [this feature request](https://features.jellyfin.org/posts/2484/add-musicbrainz-as-a-music-video-metadata-provider)?

**Comment by UltimateRiff** [comment]
on the note of music videos, many are added as just recordings, but I believe they should also have a release (and also release group), just like an audio release

**Comment by UltimateRiff** [comment]
also, adding to the confusion about track MBID and recording MBID is that MusicBrainz Picard (the official MusicBrainz tagger) adds recording MBIDs under the `MUSICBRAINZ_TRACKID` tag, and the track MBID as `MUSICBRAINZ_RELEASETRACKID` tag, because recordings used to be called tracks before track MBIDs were added

for clarity, a recording is connected to a release as a track
